packages/single-selection-mixin/single-selection-mixin.ts

```diff
@@ -0,0 +1,91 @@
+import { property, PropertyValues } from 'lit-element';
+import { DisabledStateInterface } from '@vaadin/disabled-state-mixin';
+import { KeyboardDirectionInterface } from '@vaadin/keyboard-direction-mixin';
+import { KeyboardDirectionClass } from '@vaadin/keyboard-direction-mixin/keyboard-direction-class';
+import { SelectedStateClass } from '@vaadin/selected-state-mixin/selected-state-class';
+import { SelectedStateInterface } from '@vaadin/selected-state-mixin';
+import { SlottedItemsInterface } from '@vaadin/slotted-items-mixin';
+import { SlottedItemsClass } from '@vaadin/slotted-items-mixin/slotted-items-class';
+import { SingleSelectionClass } from './single-selection-class';
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+type Constructor<T = object> = new (...args: any[]) => T;
+
+export interface SingleSelectionInterface {
+  selected: number | null | undefined;
+}
+
+const filterItems = (nodes: Node[]) => {
+  return nodes.filter((node): node is SelectedStateClass => {
+    return (
+      node.nodeType === Node.ELEMENT_NODE &&
+      (node.constructor as typeof SelectedStateClass).hasSelectedStateMixin
+    );
+  });
+};
+
+export const SingleSelectionMixin = <
+  T extends Constructor<
+    KeyboardDirectionInterface &
+      KeyboardDirectionClass &
+      SlottedItemsInterface &
+      SlottedItemsClass &
+      SingleSelectionClass
+  >
+>(
+  base: T
+): T & Constructor<SingleSelectionInterface> => {
+  class SingleSelection extends base {
+    /**
+     * The index of the item selected in the items array.
+     */
+    @property({ type: Number, reflect: true }) selected: number | null | undefined;
```

Comment:
Just a question/discussion. Don't consider this as needing changes for this PR (unless you think it's a good idea). I just want to know if you have some comments or similar ideas about this:

Now the `items` is only based on `SlottedItemsMixin` which only works with DOM elements (light DOM children) as items.

I'm wondering should we make this somehow more generic so that this could works with any kind of `items` array that might not be based on DOM elements? Should we have another "items-mixin" where `items` can be e.g. explicitly set by the user/developer to array of any data? Not sure if it would make sense (or what use cases there would be).

Or maybe the generic naming of some of these mixins is confusing me. Maybe it would be better to name this SingleElementSelectionMixin since it's only about selecting DOM elements?

Maybe it would help to have a more verbose example of what this does or what is the purpose of this mixin in the readme.

Reply:
Added the note that `SingleSelectionMixin` consumer must apply `SlottedItemsMixin`.
When not doing so, there will be a compilation error in TS. 

IMO for now this is enough to emphasize where do the items come from.